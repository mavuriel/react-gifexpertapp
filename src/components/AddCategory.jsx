import { useState } from 'react'

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState('')

  const onInputChange = ({ target: { value } }) => {
    setInputValue(value)
  }

  const onSubmit = (e) => {
    e.preventDefault()

    if (inputValue.trim().length === 0) return

    onNewCategory(inputValue)
    setInputValue('')
  }

  return (
    <form onSubmit={onSubmit}>
      <input type='text' value={inputValue} onChange={onInputChange} />
    </form>
  )
}
