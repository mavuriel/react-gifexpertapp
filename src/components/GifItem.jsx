import PropTypes from 'prop-types'

export const GifItem = ({ id, title, url }) => {
  // TODO: como cargar imagenes en redes lentas
  return (
    <div className='card'>
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  )
}

GifItem.propTypes = {
  id: PropTypes.number,
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}
