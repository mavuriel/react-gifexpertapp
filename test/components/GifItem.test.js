import { render, screen } from '@testing-library/react'
import { GifItem } from '../../src/components/GifItem'

describe('Pruebas en el componente <GifItem/>', () => {
  const title = 'Naruto'
  const url = 'https://naruto.com/hinata'

  test('debe de ser igual al snapshot', () => {
    const { container } = render(<GifItem title={title} url={url} />)

    expect(container).toMatchSnapshot()
  })

  test('debe de mostrar el URL y el alt indicado', () => {
    render(<GifItem title={title} url={url} />)

    const { src, alt } = screen.getByRole('img')

    expect(src).toBe(url)
    expect(alt).toBe(title)
  })

  test('debe mostrar el titulo en el componente', () => {
    render(<GifItem title={title} url={url} />)

    expect(screen.getByText(title)).toBeTruthy()
  })
})
