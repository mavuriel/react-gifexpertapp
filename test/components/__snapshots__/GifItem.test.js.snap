// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifItem/> debe de ser igual al snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Naruto"
      src="https://naruto.com/hinata"
    />
    <p>
      Naruto
    </p>
  </div>
</div>
`;
